mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -178,7 +179,20 @@ def encode_queries(self, queries: List[str], batch_size: int, **kwargs):
                 logger.warning(
                     "Queries will not be truncated. This could lead to memory issues. In that case please lower the batch_size."
                 )
-        return self.model.encode(queries, batch_size=batch_size, **kwargs)
+        if "instructions" in kwargs:
```

Comment:
When I was testing InstructionRetrieval with basic SentenceTransformer models I found that they didn't handle the instructions correctly and didn't like the extra kwargs.  I added this to fix that bug.

Reply:
I'd have to do a bit of digging what `kwargs["instruction"]` entails, but I think it makes sense to use the new `prompt` argument in `model.encode` here if you want to prefix something in front of all of your texts.

https://sbert.net/examples/applications/computing-embeddings/README.html#prompt-templates

Reply:
I didn't realize it had that, that's awesome!! Will add

Reply:
Ah, this actually doesn't quite work -- this setup assumes that the `prompt` is the same for the entire task (which is a fair assumption since that's what E5 and others do), whereas in retrieval with instructions, the instruction is *per query* since it's part of the data (and not part of the model). I can't see a way to give a different instruction per query, so I will leave this for now

Reply:
Yes, that would solve the issue!